### ISSUE TITLE
GPU Driver & Documentation

### ISSUE THREAD

**Issue by kamilion** [starfive-tech/VisionFive2#2]
Howdy, I've been looking through the datasheets and product briefs, and I haven't been able to find any information at all about GPU support.

All I've been able to discover so far has been
https://github.com/starfive-tech/VisionFive2/tree/JH7110_VisionFive2_devel/soft_3rdpart/IMG_GPU/out

Where is the source for this driver located?

**Comment by MichaelZhuxx** [comment]
the driver is here https://github.com/starfive-tech/linux/tree/JH7110_VisionFive2_devel/drivers/gpu/drm/img

**Comment by kamilion** [comment]
Thanks! Are the userspace support components to go along with the kernel driver also available from another repository?

**Comment by MichaelZhuxx** [comment]
yes, userspace is the link you have pointed out

**Comment by kamilion** [comment]
No, I mean, the supporting library support for accelerated graphics on top of the kernel DRM layer. X11, wayland, and vulkan? I don't recall seeing any Imagination support hitting the MESA tree recently...

 I've just spent the last ten minutes pouring over these generated register definition headers, with Imagination MIT+GPL2 headers. Looks good, and fairly complete to me; when is this generic KMS support for what appears to be ~35-40 device variants going upstream, if I may ask?
 
 Since AMD's been submitting generated register headers for quite a while now, I don't expect that to be as big of a problem as it was a decade ago, when the PowerVR work was being evaluated.
 
 
 Edit:
 Ugh, okay, I've just unpacked the tarball and discovered what I feared; all of the accelerated graphics support is binary blobs.

This is going to be a nightmare for me to get working with lubuntu's packaging...

Oh well. Off to go start writing python scripts to get ready for my board to arrive from ameridroid.

**Comment by kth5** [comment]
It seems that at least `soft_3rdpart/IMG_GPU/out/img-gpu-powervr-bin-1.17.6210866.tar.gz` on HEAD is not what the build system expects.

Reverting it to `a2c06b2c54c18961087d6caa834d6936e9862af9` helps.

It's unfortunate that we'll need blobs to drive the GPU...

**Comment by andyhu-stf** [comment]
> 

hi kth5, it also the git lfs issue, please see the issue [[](url)](https://github.com/starfive-tech/VisionFive2/issues/3)

**Comment by jackffmm** [comment]
I will not buy this device untill it has blobs.

**Comment by coderamen666** [comment]
That's on you bud. Imagination is actually working on making an open Mesa driver for 3d acceleration. The video hw acceleration is open source however.

**Comment by coderamen666** [comment]
> It seems that at least `soft_3rdpart/IMG_GPU/out/img-gpu-powervr-bin-1.17.6210866.tar.gz` on HEAD is not what the build system expects.
> 
> Reverting it to `a2c06b2c54c18961087d6caa834d6936e9862af9` helps.
> 
> It's unfortunate that we'll need blobs to drive the GPU...

It's unfortunate. Agreed. But open GPU drivers are planned to replace these blobs.

**Comment by jackffmm** [comment]
ok @jacobrogers256  but why not to release what they have now, so the development can be much faster and better ? look how the community improved the intel drivers.

**Comment by coderamen666** [comment]
> ok @jacobrogers256 but why not to release what they have now, so the development can be much faster and better ? look how the community improved the intel drivers.

I agree. It's just that IP law is draconian and the corporations take advantage of this to squeeze as much profits as possible from people while revealing as little as necessary about their products.

**Comment by aragubas** [comment]
my StarFive VisionFive 2 cannot boot StarFive's official images, it can only boot DietPi's build for the VF2. How can I install this driver on DietPi? DietPi is based on Debian
